execution/src/windows/execution.cc

```diff
@@ -117,11 +118,11 @@ namespace leatherman { namespace execution {
 
         // Format a name for the pipe based on the process and counter
         wstring name = boost::nowide::widen((boost::format("\\\\.\\Pipe\\leatherman.%1%.%2%") %
-            GetCurrentProcessId() %
+            GetCurrentThreadId() %
```

Comment:
Ah, I didn't realize overlapped I/O wasn't possible using anonymous pipes, but msdn is pretty clear on that "Asynchronous (overlapped) read and write operations are not supported by anonymous pipes."

Then I guess my only concern is what DACL is applied to the named pipe, since other processes can connect to it. But since we specify 1 as the max instance value it would only be possible between the time the pipe is created and when we connect to it (using `CreateFileW`), and if that latter one fails we raise, so :+1: